content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md

```diff
@@ -0,0 +1,207 @@
+                    unit = state, # unit index in the panel data
+                    time = year, # time index in the panel data
+                    i_unit = "California", # unit where the intervention occurred
+                    i_time = 1988, # time period when the intervention occurred
+                    generate_placebos=T # generate placebo synthetic controls (for inference)
+                    ) %>%
+  # Generate the aggregate predictors used to fit the weights (in our example referred to as W')
+  
+  # average log income, retail price of cigarettes, and proportion of the
+  # population between 15 and 24 years of age from 1980 - 1988
+  generate_predictor(time_window = 1980:1988,
+                     ln_income = mean(lnincome, na.rm = T),
+                     ret_price = mean(retprice, na.rm = T),
+                     youth = mean(age15to24, na.rm = T)) %>%
+  # average beer consumption in the donor pool from 1984 - 1988
+  generate_predictor(time_window = 1984:1988,
+                     beer_sales = mean(beer, na.rm = T)) %>%
+  
+  # Lagged cigarette sales 
+  generate_predictor(time_window = 1975,
+                     cigsale_1975 = cigsale) %>%
+  generate_predictor(time_window = 1980,
+                     cigsale_1980 = cigsale) %>%
+  generate_predictor(time_window = 1988,
+                     cigsale_1988 = cigsale) %>%
+  
+   # Generate the fitted weights for the synthetic control
+  generate_weights(optimization_window = 1970:1988, # time to use in the optimization task
+                   margin_ipop = .02,sigf_ipop = 7,bound_ipop = 6 # optimizer options
+  ) %>%
+  
+  # Generate the synthetic control
+  generate_control()
+```
+{{% /codeblock %}}
+
+Once the synthetic control is generated, one can easily assess the fit by comparing the trends of the synthetic and observed time series. The idea is that the trends in the pre-intervention period should map closely onto one another. Let's verify that with a graph: 
+
+{{% codeblock %}}
+```R
+smoking_out %>% plot_trends()
+```
+{{% /codeblock %}}
+
+<p align = "center">
+<img src = "../images/1_synthetic.png" width="700">
+</p>
+
+As you can see, the pre-intervention trend between synthetic and observed cigarette sales is the same. 
+
+To capture the causal quantity (i.e. the difference between the observed and counterfactual), one can plot the differences using plot_differences(): 
+
+{{% codeblock %}}
+```R
+smoking_out %>% plot_differences()
+```
+{{% /codeblock %}}
+
+<p align = "center">
+<img src = "../images/2_synthetic.png" width="700">
+</p>
+
+Another useful way of evaluating the synthetic control is to look at how comparable the synthetic control is to the observed covariates of the treated unit by means of the following code: 
+
+{{% codeblock %}}
+```R
+smoking_out %>% grab_balance_table()
+```
+{{% /codeblock %}}
+ 
+## Advantages and limitations 
+
+According to Abadie (2021), the advantages of synthetic control methods are: 
```

Comment:
Advantages seem a bit generic (esp point 1 and 3), some missing points:

- Sparsity: Since only a few control units are assigned non-zero weights, only a small subset out of a large pool of control units is used to construct the synthetic control. This prevents overfitting of the pre-intervention data, leading to more robust post-intervention inferences.

- No Extrapolation: Unlike some regression-based methods that may make predictions outside the range of the observed data, SCM ensures that the constructed synthetic control stays within the actual observerd outcomes of the control units. 

- Safeguard against specification searches and p-hacking: Synthetic control weights can be calculated and pre-registered before the post-treatment outcomes are realized, or before the actual intervention takes place.